django/contrib/postgres/aggregates/mixins.py

```diff
@@ -0,0 +1,50 @@
+from django.db.models.expressions import F, OrderBy
+
+
+class OrderableAggMixin:
+
+    def __init__(self, expression, ordering=(), **extra):
+        if not isinstance(ordering, (list, tuple)):
+            ordering = [ordering]
+        ordering = ordering or []
+        # Transform minus sign prefixed strings into an OrderBy() expression.
+        ordering = [
+            (OrderBy(F(o[1:]), descending=True) if isinstance(o, str) and o[0] == '-' else o)
+            for o in ordering
+        ]
+        super().__init__(expression, **extra)
+        self.ordering = self._parse_expressions(*ordering)
+
+    def resolve_expression(self, *args, **kwargs):
+        self.ordering = [expr.resolve_expression(*args, **kwargs) for expr in self.ordering]
+        return super().resolve_expression(*args, **kwargs)
+
+    def as_sql(self, compiler, connection):
+        # Turn ordering parameter to ORDER BY SQL clause.
+        if self.ordering:
+            ordering_sql = [ordering_element.as_sql(compiler, connection)[0] for ordering_element in self.ordering]
+            self.extra['ordering'] = 'ORDER BY ' + ', '.join(ordering_sql)
+        else:
+            self.extra['ordering'] = ''
+        return super().as_sql(compiler, connection)
+
+    def get_source_expressions(self):
+        return self.source_expressions + self.ordering
+
+#    def set_source_expressions(self, exprs):
```

Comment:
I don't see any tests failing with this method removed. Is it required? If so, could you add a test to show why?